test/NuGet.Core.Tests/NuGet.PackageManagement.Test/NuGetPackageManagerTests.cs

```diff
@@ -4768,6 +4768,693 @@ public Task TestPacManGetUnrestoredPackagesByDependencyOrder()
+
+                // Assert Project1
+                // Check that the packages.config file exists after the installation
+                Assert.True(File.Exists(packagesConfigPath1));
+
+                // Check the number of packages and packages returned by PackagesConfigProject after the installation
+                var packagesInPackagesConfig = (await msBuildNuGetProject1.PackagesConfigNuGetProject.GetInstalledPackagesAsync(token)).ToList();
+                Assert.Equal(1, packagesInPackagesConfig.Count);
+                Assert.Equal(packageIdentity, packagesInPackagesConfig[0].PackageIdentity);
+                Assert.NotEmpty(firstBatchStartId);
+                Assert.NotEmpty(firstBatchEndId);
+                Assert.Equal(firstBatchStartId, firstBatchEndId);
+
+                // check project name
+                string firstProjectNameExpected;
+                msBuildNuGetProject1.TryGetMetadata(NuGetProjectMetadataKeys.Name, out firstProjectNameExpected);
+                Assert.Equal(firstProjectNameExpected, firstProjectName);
+
+                // second project
+                var msBuildNuGetProject2 = testSolutionManager.AddNewMSBuildProject();
+                var packagesConfigPath2 = msBuildNuGetProject2.PackagesConfigNuGetProject.FullPath;
+
+                var secondBatchStartId = string.Empty;
+                var secondBatchEndId = string.Empty;
+                var secondProjectName = string.Empty;
+
+                // add batch events handler
+                nuGetPackageManager.BatchStart += (o, args) =>
+                {
+                    secondBatchStartId = args.Id;
+                    secondProjectName = args.Name;
+                };
+
+                nuGetPackageManager.BatchEnd += (o, args) =>
+                {
+                    secondBatchEndId = args.Id;
+                };
+
+                // Act
+                await nuGetPackageManager.InstallPackageAsync(msBuildNuGetProject2, packageIdentity,
+                    new ResolutionContext(), new TestNuGetProjectContext(), sourceRepositoryProvider.GetRepositories().First(), null, token);
+
+                // Assert Project2
+                // Check that the packages.config file exists after the installation
+                Assert.True(File.Exists(packagesConfigPath2));
+
+                // Check the number of packages and packages returned by PackagesConfigProject after the installation
+                packagesInPackagesConfig = (await msBuildNuGetProject2.PackagesConfigNuGetProject.GetInstalledPackagesAsync(token)).ToList();
+                Assert.Equal(1, packagesInPackagesConfig.Count);
+                Assert.Equal(packageIdentity, packagesInPackagesConfig[0].PackageIdentity);
+                Assert.NotEmpty(secondBatchStartId);
+                Assert.NotEmpty(secondBatchEndId);
+                Assert.Equal(secondBatchStartId, secondBatchEndId);
```

Comment:
Where is the check to verify that the batch id changed between the first and second installs? What is the point of this test?